### PR TITLE
Implement ListValidatorAssignments RPC function

### PR DESCRIPTION
**Pull request by terencechain**
Part of #3045 

This PR implements the `ListValidatorAssignments ` functionality of the `BeaconChainService`.

Changed:
- Refactored state and end page calculation into it's own `startAndEndPage`
- Implemented `ListValidatorAssignments` RPC function
- Tests `ListValidatorAssignments` RPC function via the following:
  - List assignments fail when exceeds max page size
  - List assignments fail when requested index goes out of range
  - List assignments pass with default page size while no pubkey/index specified
  - List assignments pass with options pubkey and index specified and no page size
  - List assignments pass with options and page size specified
  - List assignments pass with options and page size specified in wrap around scenario

### CODE REVIEW

File: beacon-chain/rpc/beacon_chain_server.go [hunk 1]
```diff
@@ -215,3 +305,35 @@ func (bs *BeaconChainServer) GetValidatorParticipation(
 ) (*ethpb.ValidatorParticipation, error) {
 	return nil, status.Error(codes.Unimplemented, "not implemented")
 }
+
+// startAndEndPage is a pagination heloer, it takes in the requested page token, size, total size,
```
Comment (rauljordan) [inline beacon-chain/rpc/beacon_chain_server.go]:
```suggestion
// startAndEndPage is a pagination helper, it takes in the requested page token, size, total size,
```

File: beacon-chain/rpc/beacon_chain_server.go [hunk 2]
```diff
@@ -215,3 +305,35 @@ func (bs *BeaconChainServer) GetValidatorParticipation(
...
+// and returns start, end page and the next page token.
+func (bs *BeaconChainServer) startAndEndPage(pageToken string, pageSize int, totalSize int) (int, int, string, error) {
```
Comment (rauljordan) [inline beacon-chain/rpc/beacon_chain_server.go]: This doesn't need to be a pointer receiver as it is a pretty pure function. Can this go into some util like maybe a pagination.go file or shared/pagination?

Comment (terencechain) [reply beacon-chain/rpc/beacon_chain_server.go]: Good feedback. Done

File: beacon-chain/rpc/beacon_chain_server.go [hunk 3]
```diff
@@ -196,14 +175,125 @@ func (bs *BeaconChainServer) GetValidatorQueue(
...
 //
-// This request may specify optional validator indices or public keys to
-// filter validator assignments.
+// TODO(#3045): Implement validator set for a specific epoch. Current implementation returns latest set,
+// this is blocked by DB refactor.
 func (bs *BeaconChainServer) ListValidatorAssignments(
 	ctx context.Context, req *ethpb.ListValidatorAssignmentsRequest,
 ) (*ethpb.ValidatorAssignments, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	if int(req.PageSize) > params.BeaconConfig().MaxPageSize {
```
Comment (nisdas) [inline beacon-chain/rpc/beacon_chain_server.go]: could you break it into simpler functions ? maybe handling the request with a filter as one function 
and providing the default response as another

**Pull request merged**